enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,398 @@
+We will define a new value for the `TopologyMode` enum: `DualReplicaTopologyMode`.
+
+However `TopologyMode` is not available at the point the Agent Based Installer (ABI) performs validation.
+We will therefore additionally define a new feature gate `DualReplicaTopology` that can be enabled in `install-config.yaml`, and which ABI can use to validate the proposed cluster - such as the proposed node count.
+
+#### CEO Trigger
+
+Initially the creation of an etcd cluster will be driven in the same way as other platforms.
+Once the cluster has two members, the etcd daemon will be removed from the static pod definition and recreated as a resource controlled by RHEL-HA.
+At this point, the Cluster Etcd Operator (CEO) will be made aware of this change so that some membership management functionality that is now handled by RHEL-HA can be disabled.
+This will be achieved by having the same entity that drives the configuration of RHEL-HA use the OpenShift API to update a field in the CEO's `ConfigMap` - which can only succeed if the control-plane is healthy.
+
+To enable this flow, we propose the addition of a `managedEtcdKind` field which defaults to `Cluster` but will be set to `External` during installation, and will only be respected if the `Infrastructure` CR's `TopologyMode` is `DualReplicaTopologyMode`.
+This will allow the use of a credential scoped to `ConfigMap`s in the `openshift-etcd-operator` namespace, to make the change.
+
+### Topology Considerations
+
+2NO represents a new topology, and is not appropriate for use with HyperShift, SNO, or MicroShift
+
+#### Standalone Clusters
+
+Is the change relevant for standalone clusters?
+TODO: Exactly what is the definition of a standalone cluster?  Disconnected?  Physical hardware?
+
+### Implementation Details/Notes/Constraints
+
+While the target installation requires exactly 2 nodes, this will be achieved by building support in the core installer for a "bootstrap plus 2 nodes" flow, and then using Assisted Installer's ability to bootstrap-in-place to remove the requirement for a bootstrap node.
+
+The delivery of RHEL-HA components will be opaque to the user and be delivered as an [MCO Extension](../rhcos/extensions.md) in the 4.18 and 4.19 timeframes.
+A switch to [MCO Layering](../ocp-coreos-layering/ocp-coreos-layering.md ) will be investigated once it is GA in a shipping version of OpenShift.
+
+Configuration of the RHEL-HA components will be via one or more `MachineConfig`s, and will require RedFish details to have been collected by the installer.
+Sensible defaults will be chosen where possible, and user customization only where absolutely necessary.
+
+The entity (likely a one-shot systemd job as part of a `MachineConfig`) that configures RHEL-HA will also configure a fencing priority.
+This is usually done based on the sort-order a piece of shared info (such as IP or node name).
```

Comment:
Can we be specific about how this will be achieved? You say usually here, but, what would you implement it as here?

Is this just a case of stating the node that comes first alphabetically will wait 10s and the node that comes second will wait 20?

Just trying to think more concretely about this where the current description is quite theoretical still

Reply:
more like 0s and 20s, but yes, alphabetical order is both the sanest and easiest approach, so that's what I'd recommend.

Reply:
Are we OK with moving forward with this proposal?